client/client.go

```diff
@@ -12,63 +12,53 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// Package client contains generic representations of clients connecting to different receivers
+// Package client contains generic representations of clients connecting to different receivers. Components,
+// such as processors or exporters, can make use of this information to make decisions related to
+// grouping of batches, tenancy, load balancing, tagging, among others.
 package client // import "go.opentelemetry.io/collector/client"
 
 import (
 	"context"
 	"net"
-	"net/http"
-
-	"google.golang.org/grpc/peer"
 )
 
 type ctxKey struct{}
 
-// Client represents a generic client that sends data to any receiver supported by the OT receiver
-type Client struct {
-	IP string
-}
+// Info contains data related to the clients connecting to receivers.
+type Info struct {
+	// IP for the client connecting to this collector. Available in a best-effort basis, and generally reliable
+	// for receivers making use of confighttp.ToServer and configgrpc.ToServerOption.
+	IP *net.IPAddr
```

Comment:
I don't think we can make this assumption that we always use IP, we can have unix sockets correct? So this really needs to be the interface net.Addr as mentioned in the previous comment.

Reply:
Good point about Unix sockets!